### ISSUE TITLE
MSSQL attempting to use LIMIT expression instead of TOP

### ISSUE THREAD

**Issue by JakeAM** [gogits/gogs#4176]
- Gogs version (or commit ref): 0.10 RC
- Git version: 2.8.1.windows.1
- Operating system: Windows Server 2012
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] SQLite
  - [X] MSSQL
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist (usually found in `log/gogs.log`): https://gist.github.com/JakeAM/8fdecbf4aadb32d945616aabdf034cb4

## Description

Unless you have admin privileges, when you try and access an organisation it throws the error found in the gist above. I have investigated it with SQL profiler and it turns out it's trying to use the LIMIT keyword which doesn't exist in SQL Server. Below is the query it tries to run if the user is not an admin.

```
SELECT repository.* FROM repository
	INNER JOIN team_repo 
	ON team_repo.repo_id = repository.id
	WHERE (repository.owner_id = 1 AND repository.is_private = 0) OR team_repo.team_id IN (5)
	GROUP BY repository.id
	ORDER BY updated_unix DESC
	LIMIT 15 OFFSET 0
```

I would guess this is actually an issue with underlying ORM?

**Comment by Unknwon** [comment]
Thanks, didn't know that MSSQL doesn't have `limit`.

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging #4194, please help test on `develop` branch.

**Comment by Unknwon** [comment]
Please notify if fix doesn't work, closing now.
